_components/sockets_component.h

```diff
@@ -58,11 +60,18 @@ void socket_transmitter_sta_loop(bool (*is_wifi_connected)()) {
             }
 
 #ifdef CONFIG_PACKET_RATE
-            vTaskDelay(CONFIG_PACKET_RATE != 0 ? floor(1000 / CONFIG_PACKET_RATE) : 0);
-            ets_delay_us(((1000.0 / CONFIG_PACKET_RATE) - floor(1000 / CONFIG_PACKET_RATE)) * 1000);
+            if(CONFIG_PACKET_RATE > 0) {
+                double wait_duration = (1000.0 / CONFIG_PACKET_RATE) - lag;
+                int w = floor(wait_duration);
+                vTaskDelay(w);
+                // ets_delay_us((wait_duration - w) * 1000);
```

Comment:
Should this comment be deleted? Or should it be uncommented?

Reply:
Alternatively, we can leave this commented out and add a comment if `ets_delay_us` is not very accurate (for example).

Reply:
I am omitting this commented line because it is making a drop for 200Hz Tx rate (195-197Hz achieved with this line active).